libraries/chain/include/eosio/chain/apply_context.hpp

```diff
@@ -585,6 +585,9 @@ class apply_context {
       int32_t  kv_it_lower_bound(uint32_t itr, const char* key, uint32_t size, uint32_t* found_key_size, uint32_t* found_value_size);
       int32_t  kv_it_key(uint32_t itr, uint32_t offset, char* dest, uint32_t size, uint32_t& actual_size);
       int32_t  kv_it_value(uint32_t itr, uint32_t offset, char* dest, uint32_t size, uint32_t& actual_size);
+      kv_context& kv_get_backing_store() const {
```

Comment:
Is kv_context an immutable type?  Or should we return a const reference since this method is marked const?

Reply:
It is a mutable type. Could not return a const reference.

Reply:
Then should this method be invocable on a const instance?

Reply:
This method is invoked on apply_context, which is not constant. Its use looks like "int64_t apply_context::kv_set(...) {
 return kv_get_backing_store().kv_set(...);" 

If I returned a const reference, I got a compile error:

/home/lin/Work/kv/eos/libraries/chain/apply_context.cpp:1140:11: error:
'this' argument to member function 'kv_set' has type 'const
      eosio::chain::kv_context', but function is not marked const
   return kv_get_backing_store().kv_set"

Reply:
What I was thinking is that if the method is marked const, then we might want to return a const refererence.  Otherwise, if the method doesn't need to be constant, then can we remove the const from the method?

Reply:
Oh I see your point. I'll remove the const. Thanks.